File: csharp/Primrose.NETStandard2/Grammars/PlainText.cs
using System;
using System.Text.RegularExpressions;

namespace Primrose
{
    public partial class Grammar
    {
        /// <summary>
        /// A grammar that makes displaying plain text work with the text editor designed for syntax highlighting.
        /// </summary>
        public static readonly Grammar PlainText = new Grammar("PlainText", Array.Empty<(string, Regex)>());
    }
}

File: js/package/src/Grammars/PlainText.js
/*
pliny.value({
  parent: "Primrose.Text.Grammars",
  name: "PlainText",
  description: "A grammar that makes displaying plain text work with the text editor designed for syntax highlighting."
});
*/

import { Grammar } from "./Grammar.js";
export const PlainText = new Grammar("PlainText", [
    ["newlines", /(?:\r\n|\r|\n)/],
    ["whitespace", /(?:\s+)/]
]);
